celery/app/utils.py

```diff
@@ -84,31 +84,31 @@ def __init__(self, *args, deprecated_settings=None, **kwargs):
     @property
     def broker_read_url(self):
         return (
-            os.environ.get('CELERY_BROKER_READ_URL') or
             self.get('broker_read_url') or
+            os.environ.get('CELERY_BROKER_READ_URL') or
             self.broker_url
         )
 
     @property
     def broker_write_url(self):
         return (
-            os.environ.get('CELERY_BROKER_WRITE_URL') or
             self.get('broker_write_url') or
+            os.environ.get('CELERY_BROKER_WRITE_URL') or
             self.broker_url
```

Comment:
This behaviour for `broker_read_url` / `broker_write_url` may be surprising, as I think it's less clear cut whether an explicitly defined `broker_url` should be used ahead of specific env vars for read/write URLs.

In this case, I think that env vars for read/write URLs are probably more specific than explicit setting for `broker_url`, so I set the priority accordingly. 

Happy to adjust if necessary

Reply:
@soceanainn Thank you for your contribution. 

I think this might however break a lot of Django projects with stray environment variables, so this should be announced in some change log I think, not sure if it needs to be part of this PR.

And also we should make this super explicit in the docs.